src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -626,6 +626,11 @@ CreateHashDistributedTable(Oid relationId, char *distributionColumnName,
 	 * Get an exclusive lock on the colocation system catalog. Therefore, we
 	 * can be sure that there will no modifications on the colocation table
 	 * until this transaction is committed.
```

Comment:
The "until this transaction is committed" part of this comment is no longer accurate. 

I think the important point to get across is that we take an `ExclusiveLock` in case the colocation group does not exist yet and we want to avoid concurrent `create_distributed_table` calls creating 2 separate colocation groups with the same parameters. Then if the colocation group already exists we can relax the locking since we're not going to modify `pg_dist_colocation`, and otherwise we keep it such that concurrent `create_distributed_table` calls will block and then see the new colocation group.